mlflow/utils/process.py

```diff
@@ -56,21 +62,35 @@ def _exec_cmd(
         raise ValueError("`extra_env` and `env` cannot be used at the same time")
 
     env = env if extra_env is None else {**os.environ, **extra_env}
-    prc = subprocess.run(
-        # In Python < 3.8, `subprocess.Popen` doesn't accpet a command containing path-like
-        # objects (e.g. `["ls", pathlib.Path("abc")]`) on Windows. To avoid this issue,
-        # stringify all elements in `cmd`. Note `str(pathlib.Path("abc"))` returns 'abc'.
-        map(str, cmd),
-        env=env,
-        check=False,
-        capture_output=capture_output,
-        text=True,
-        **kwargs,
-    )
-
-    if throw_on_error and prc.returncode != 0:
-        raise ShellCommandException.from_completed_process(prc)
-    return prc
+
+    # In Python < 3.8, `subprocess.Popen` doesn't accpet a command containing path-like
+    # objects (e.g. `["ls", pathlib.Path("abc")]`) on Windows. To avoid this issue,
+    # stringify all elements in `cmd`. Note `str(pathlib.Path("abc"))` returns 'abc'.
+    cmd = list(map(str, cmd))
+
+    if synchronous:
+        prc = subprocess.run(
+            cmd,
+            env=env,
+            check=False,
+            capture_output=capture_output,
+            text=True,
+            **kwargs,
+        )
+        if throw_on_error and prc.returncode != 0:
+            raise ShellCommandException.from_completed_process(prc)
+        return prc
+    else:
+        if capture_output:
+            kwargs["stdout"] = subprocess.PIPE
+            kwargs["stderr"] = subprocess.PIPE
```

Comment:
```suggestion
        if capture_output:
            if kwargs.get('stdout') is not None or kwargs.get('stderr') is not None:
                raise ValueError('stdout and stderr arguments may not be used '
                                 'with capture_output.')
            kwargs["stdout"] = subprocess.PIPE
            kwargs["stderr"] = subprocess.PIPE
```

Let's use the same validation as `subprocess`

https://github.com/python/cpython/blob/0897a0bf9c75bacf428b8f6f08114fffc9f9c542/Lib/subprocess.py#L495